patches/minecraft/net/minecraft/world/entity/ai/behavior/StartAttacking.java.patch

```diff
@@ -0,0 +1,16 @@
+--- a/net/minecraft/world/entity/ai/behavior/StartAttacking.java
++++ b/net/minecraft/world/entity/ai/behavior/StartAttacking.java
+@@ -46,7 +_,11 @@
+    }
+ 
+    public static <E extends Mob> void m_24213_(E p_24214_, LivingEntity p_24215_) {
+-      p_24214_.m_6274_().m_21879_(MemoryModuleType.f_26372_, p_24215_);
+-      p_24214_.m_6274_().m_21936_(MemoryModuleType.f_26326_);
++      net.minecraftforge.event.entity.living.LivingChangeTargetEvent changeTargetEvent = net.minecraftforge.common.ForgeHooks.onLivingChangeTarget(p_24214_, p_24215_, net.minecraftforge.event.entity.living.LivingChangeTargetEvent.LivingTargetType.BEHAVIOR_TARGET);
++      if(!changeTargetEvent.isCanceled()) {
++          p_24214_.m_6274_().m_21879_(MemoryModuleType.f_26372_, changeTargetEvent.getNewTarget());
++          p_24214_.m_6274_().m_21936_(MemoryModuleType.f_26326_);
++          net.minecraftforge.common.ForgeHooks.onLivingSetAttackTarget(p_24214_, changeTargetEvent.getNewTarget(), net.minecraftforge.event.entity.living.LivingChangeTargetEvent.LivingTargetType.BEHAVIOR_TARGET);
```

Comment:
Since this is being marked for removal, should add a `TODO: Remove 1.20`

Reply:
Just to reassure that I understood you correctly: Should I add that comment to the patch (or really any occurrence of firing the `LivingSetAttackTargetEvent`) *and* the event class or should I add this comment just to the event class?

Reply:
I also have a little style question. When adding that normal comment to the LivingSetAttackTargetEvent, should I put it above the JavaDoc or beneath it?

Reply:
> Just to reassure that I understood you correctly: Should I add that comment to the patch (or really any occurrence of firing the LivingSetAttackTargetEvent) and the event class or should I add this comment just to the event class?

I would add the comment to the event class and calls to the event class since it's easier to double check that everything should be removed.

> I also have a little code style question. When adding that normal comment to the LivingSetAttackTargetEvent, should I put it above the JavaDoc or beneath it?

Same line as the deprecation annotation imo.

Reply:
Ok, I've now added the TODOs and the deprecations with commit 6c510de6a5867b7c612eef839fbd7874dce9fcaf.